src/modules/0.0.22/aws_vpc/mappers/subnet.ts

```diff
@@ -9,27 +9,28 @@ export class SubnetMapper extends MapperBase<Subnet> {
   module: AwsVpcModule;
   entity = Subnet;
   equals = (a: Subnet, b: Subnet) =>
-    Object.is(a.subnetId, b.subnetId) && Object.is(a?.availabilityZone?.name, b?.availabilityZone?.name); // TODO: Do better
+    Object.is(a.cidrBlock, b.cidrBlock) &&
+    Object.is(a?.availabilityZone?.name, b?.availabilityZone?.name) &&
+    Object.is(a?.vpc?.vpcId, b?.vpc?.vpcId);
 
-  async subnetMapper(sn: AwsSubnet, ctx: Context) {
+  async subnetMapper(sn: AwsSubnet, ctx: Context, region: string) {
     const out = new Subnet();
     if (!sn?.SubnetId || !sn?.VpcId) return undefined;
     out.state = sn.State as SubnetState;
     if (!sn.AvailabilityZone) return undefined;
     out.availabilityZone =
-      (await this.module.availabilityZone.db.read(ctx, sn.AvailabilityZone)) ??
-      (await this.module.availabilityZone.cloud.read(ctx, sn.AvailabilityZone));
+      (await this.module.availabilityZone.db.read(ctx, `${sn.AvailabilityZone}|${region}`)) ??
```

Comment:
nit mentioned this in another PR, but having a static getter/setter function in the cloud id service that gets called here would be nice

Reply:
Agreed

Reply:
Well it would need to be like a helper like createCompositeKey or something customizable. In this case we couldn't used the entityId because the values come from different objects.